### PR TITLE
851 add network configuration file and refactor to use NetworkManager

### PR DESCRIPTION
**Pull request by B-Naoufal**
FLag for node server will be deprecated and remove just after the review.

### CODE REVIEW

File: pkg/config/network.go [hunk 1]
```diff
@@ -1,71 +1,165 @@
...
-		return dnsFlag
+func (n *NetworkManager) SetNetwork(config AppConfig) {
+	n.networkMutex.Lock()
+	defer n.networkMutex.Unlock()
+
+	n.appConfig = config
+}
+
+func (n *NetworkManager) Network() *AppConfig {
+	n.networkMutex.RLock()
```
Comment (peterjah) [inline pkg/config/network.go:122]: mutex is not needed here!

**Review by peterjah — CHANGES_REQUESTED** [review]

**Comment by peterjah** [comment]
names of structures are not well chosen. You create a networkManager who contains an AppConfig.
I would have expected the opposite. an ConfigManager who contains a networkConfig, and maybe other configs

File: pkg/config/network.go [hunk 2]
```diff
@@ -1,71 +1,165 @@
...
+	}
+
+	return &options
 }
 
-func GetNodeURL(urlOrNetwork string) string {
-	switch urlOrNetwork {
-	case "TESTNET":
-		return testnetNodeURL
+func (n *NetworkManager) NetworkFromString(optionStr string) (*AppConfig, error) {
```
Comment (Thykof) [inline pkg/config/network.go:91]: argument optionStr is badly named, should it be networkName?

File: pkg/config/network.go [hunk 3]
```diff
@@ -1,71 +1,165 @@
...
+	networkManager.SetNetwork(appConfig)
+
+	return networkManager, nil
+}
+
+func (n *NetworkManager) SetNetworks(networks map[string]NetworkConfig) {
+	n.knownNetworks = networks
+}
+
+func (n *NetworkManager) Networks() *[]string {
```
Comment (Thykof) [inline pkg/config/network.go:81]: maybe this function should be name KnownNetworksName?

**Review by Thykof — CHANGES_REQUESTED** [review]
please run task fmt to format and lint the code.

**Comment by B-Naoufal** [comment]
> AppConfig

since massaSTATION use AppConfig everywhere , i prefer to open a seperate PR for only renaming the configmanager and appConfig an keep the diffs in this PR clean 
image
@peterjah what do you think ??

File: pkg/config/network.go (hunk 3 above)
Comment (B-Naoufal) [reply pkg/config/network.go:81]: i think its better to have short names and the comment is very explicit here : 	// Networks retrieves a pointer to a slice of known networks from the NetworkManager.
	// It returns a pointer to a slice containing the names of the known networks.
	// The slice will be updated if the known networks are modified.
	Networks() *[]string

File: pkg/config/network.go [hunk 4]
```diff
@@ -1,71 +1,168 @@
...
+	knownNetworks map[string]NetworkConfig // Known network configurations
+	networkMutex  sync.RWMutex             // Mutex for concurrent access to network configuration
+}
 
-	buildnetNodeURL    = "https://buildernet.massa.net/api/v2"
-	buildnetDNSAddress = "AS12aGWkBorEM2EpKeNyigSkoCqwdxm872g5KkzUiox3v4VosFW3F"
+// Verify at compilation time that NetworkManager implements NetworkManagerInterface.
+//
+//nolint:exhaustruct
+var _ NetworkManagerInterface = &NetworkManager{}
```
Comment (peterjah) [inline pkg/config/network.go:26]: why do you need an interface ? :/ from my experience, we use it when we need to mock the object. but here i don't see the purpose

File: pkg/config/interface.go [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+
+	// Network returns the current network configuration.
+	// It returns the network configuration represented by an AppConfig struct.
+	Network() *AppConfig
+
+	// SwitchNetwork switches the current network configuration to the specified network.
+	// selectedNetworkStr: The name of the network configuration to switch to.
+	// Returns any error encountered during the switch operation.
+	SwitchNetwork(selectedNetworkStr string) error
+}
```
Comment (peterjah) [inline pkg/config/interface.go:44]: I don't think you need an interface

File: pkg/config/network.go (hunk 1 above)
Comment (peterjah) [reply pkg/config/network.go:122]: i unresolve this because you did not removed the mutex

**Review by peterjah — CHANGES_REQUESTED** [review]

**Comment by peterjah** [comment]
> > AppConfig
> 
> since massaSTATION use AppConfig everywhere , i prefer to open a seperate PR for only renaming the configmanager and appConfig an keep the diffs in this PR clean @peterjah what do you think ??

Ok. but imho you are taking the problem in the wrong way.
you should keep AppCOnfig. but rename what you added from "networkManager" to "configManager" to be more generic..

**Review by Thykof — COMMENTED** [review]
OK for me if Pierre's comment are solved

**Review by peterjah — APPROVED** [review]

**Review by Thykof — APPROVED** [review]

**Pull request merged**